code/game/gamemodes/trifecta/trifecta.dm

```diff
@@ -0,0 +1,116 @@
+#define TOT_COST 5
+#define VAMP_COST 10
+#define CLING_COST 10
+
+
+/datum/game_mode/trifecta
+	name = "Trifecta"
+	config_tag = "trifecta"
+	protected_jobs = list("Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Blueshield", "Nanotrasen Representative", "Magistrate", "Internal Affairs Agent", "Nanotrasen Navy Officer", "Special Operations Officer", "Solar Federation General")
+	restricted_jobs = list("Cyborg")
+	secondary_restricted_jobs = list("AI")
+	required_players = 10
+	required_enemies = 1	// how many of each type are required
+	recommended_enemies = 3
+	secondary_protected_species = list("Machine")
+	var/vampire_restricted_jobs = list("Chaplain")
+	var/list/datum/mind/pre_traitors = list()
+	var/list/datum/mind/pre_changelings = list()
+	var/list/datum/mind/pre_vampires = list()
+	var/amount_vamp = 1
+	var/amount_cling = 1
+	var/amount_tot = 1
+
+/datum/game_mode/trifecta/announce()
+	to_chat(world, "<b>The current game mode is - Trifecta</b>")
+	to_chat(world, "<b>Vampires, traitors, and changelings, oh my! Stay safe as these forces work to bring down the station.</b>")
+
+
+/datum/game_mode/trifecta/pre_setup()
+	calculate_quantities()
+	if(GLOB.configuration.gamemode.prevent_mindshield_antags)
+		restricted_jobs += protected_jobs
+	var/list/datum/mind/possible_vampires = get_players_for_role(ROLE_VAMPIRE)
+
+	if(!length(possible_vampires))
+		return FALSE
+
+	for(var/I in possible_vampires)
+		if(length(pre_vampires) >= amount_vamp)
+			break
+		var/datum/mind/vampire = pick_n_take(possible_vampires)
+		if(vampire.current.client.prefs.active_character.species in secondary_protected_species)
+			continue
+		pre_vampires += vampire
+		vampire.special_role = SPECIAL_ROLE_VAMPIRE
+		vampire.restricted_roles = (restricted_jobs + secondary_restricted_jobs + vampire_restricted_jobs)
+
+	//Vampires made, off to changelings
+	var/list/datum/mind/possible_changelings = get_players_for_role(ROLE_CHANGELING)
+
+	if(!length(possible_changelings))
+		return FALSE
+
+	for(var/I in possible_changelings)
+		if(length(pre_changelings) >= amount_cling)
+			break
+		var/datum/mind/changeling = pick_n_take(possible_changelings)
+		if((changeling.current.client.prefs.active_character.species in secondary_protected_species) || changeling.special_role == SPECIAL_ROLE_VAMPIRE)
+			continue
+		pre_changelings += changeling
+		changeling.restricted_roles = (restricted_jobs + secondary_restricted_jobs)
+		changeling.special_role = SPECIAL_ROLE_CHANGELING
+
+	//And now traitors
+	var/list/possible_traitors = get_players_for_role(ROLE_TRAITOR)
+
+	//stop setup if no possible traitors
+	if(!length(possible_traitors))
+		return FALSE
+
+	for(var/i in 1 to amount_tot)
```

Comment:
This code works slightly different from the above ones. Doing it this way can mean that you won't have `amount_tot` amount of tots